executor/partition_table.go

```diff
@@ -107,6 +114,33 @@ func nextPartitionWithTrace(ctx context.Context, n nextPartition, tbl table.Phys
 	return n.nextPartition(ctx, tbl)
 }
 
+// updateDAGRequestTableID update the table ID in the DAG request to partition ID.
+// TiKV only use that table ID for log, but TiFlash use it.
+func updateDAGRequestTableID(ctx context.Context, dag *tipb.DAGRequest, tableID, partitionID int64) {
+	// TiFlash set RootExecutor field and ignore Executors field.
+	if dag.RootExecutor != nil {
+		updateExecutorTableID(ctx, dag.RootExecutor, tableID, partitionID)
```

Comment:
root executor means the oldest ancestor , which is not always datasource.
In this case ,you should check leaf node